pkg/container-collection/options.go

```diff
@@ -517,64 +517,69 @@ func WithKubernetesEnrichment(nodeName string, kubeconfig *rest.Config) Containe
+						return false
+					}
+				} else {
+					pod, err = clientset.CoreV1().Pods(container.K8s.Namespace).Get(context.TODO(), container.K8s.PodName, metav1.GetOptions{})
+					if err != nil {
+						log.Errorf("kubernetes enricher (from ns/podname): cannot find pod %s/%s: %s", container.K8s.Namespace, container.K8s.PodName, err)
+						return false
+					}
 				}
-			}
 
-			if container.K8s.ContainerName == "" {
-				var containerName string
-				uid := string(pod.ObjectMeta.UID)
-				containerNames := []string{}
-				for _, c := range pod.Spec.Containers {
-					containerNames = append(containerNames, c.Name)
-				}
-				for _, c := range pod.Spec.InitContainers {
-					containerNames = append(containerNames, c.Name)
-				}
-				for _, c := range pod.Spec.EphemeralContainers {
-					containerNames = append(containerNames, c.Name)
-				}
-			outerLoop:
-				for _, name := range containerNames {
-					for _, m := range container.OciConfig.Mounts {
-						pattern := fmt.Sprintf("pods/%s/containers/%s/", uid, name)
-						if strings.Contains(m.Source, pattern) {
-							containerName = name
-							break outerLoop
+				if container.K8s.ContainerName == "" {
+					var containerName string
+					uid := string(pod.ObjectMeta.UID)
+					containerNames := []string{}
+					for _, c := range pod.Spec.Containers {
+						containerNames = append(containerNames, c.Name)
+					}
+					for _, c := range pod.Spec.InitContainers {
+						containerNames = append(containerNames, c.Name)
+					}
+					for _, c := range pod.Spec.EphemeralContainers {
+						containerNames = append(containerNames, c.Name)
+					}
+				outerLoop:
+					for _, name := range containerNames {
+						for _, m := range container.OciConfig.Mounts {
+							pattern := fmt.Sprintf("pods/%s/containers/%s/", uid, name)
+							if strings.Contains(m.Source, pattern) {
+								containerName = name
+								break outerLoop
+							}
 						}
 					}
+					container.K8s.ContainerName = containerName
 				}
-				container.K8s.ContainerName = containerName
-			}
 
-			container.K8s.Namespace = pod.ObjectMeta.Namespace
-			container.K8s.PodName = pod.ObjectMeta.Name
-			container.K8s.PodUID = string(pod.ObjectMeta.UID)
-			container.K8s.PodLabels = pod.ObjectMeta.Labels
+				container.K8s.Namespace = pod.ObjectMeta.Namespace
+				container.K8s.PodName = pod.ObjectMeta.Name
+				container.K8s.PodUID = string(pod.ObjectMeta.UID)
+				container.K8s.PodLabels = pod.ObjectMeta.Labels
 
-			// drop pause containers
-			if container.K8s.PodName != "" && container.K8s.ContainerName == "" {
-				return false
+				// drop pause containers
+				if container.K8s.PodName != "" && container.K8s.ContainerName == "" {
+					return false
+				}
 			}
 
+			if container.K8s.ownerReference == nil {
+				_, err = container.GetOwnerReference()
+				if err != nil {
+					log.Errorf("kubernetes enricher: failed to get owner reference for container %s: %s", container.Runtime.ContainerID, err)
+					return false
```

Comment:
`return false`: Should the container really be dropped if fetching the owner from the api server fails?

Reply:
That is a good question. I think `GetOwnerReference` only failys if there are probablems with the API server. So I would say that is an error worth reporting.

But is dropping then the right choice? Idk

Reply:
I would keep the `log.Errorf` but remove the `return false`.

If users run `kubectl delete replicaset foo --force`, then the replica set will be deleted immediately, so the call to the api server will fail. But the processes in the container will not be terminated immediately, so we still need to keep the container in the container collection to enrich ebpf events.